x-pack/platform/plugins/shared/rule_registry/server/alert_data_client/alerts_client.ts

```diff
@@ -1228,4 +1243,77 @@ export class AlertsClient {
       fields,
     };
   }
+
+  public getRuleTypeIds(ruleTypeIds: string | string[] | undefined): string[] {
+    // fetch all rule types if no specific rule type Ids are provided
+    if (!ruleTypeIds || ruleTypeIds.length === 0) {
+      const registeredRuleTypes = this.getRuleList();
+
+      if (!registeredRuleTypes) {
+        throw Boom.badRequest('No rule types are registered');
+      }
+
+      return Array.from(registeredRuleTypes.keys());
+    }
+
+    return Array.isArray(ruleTypeIds) ? ruleTypeIds : [ruleTypeIds];
+  }
+
+  public async getAlertFields(
+    ruleTypeIds: string | string[]
+  ): Promise<{ alertFields: BrowserFields; fields: FieldDescriptor[] }> {
+    const allRuleTypesIds = this.getRuleTypeIds(ruleTypeIds);
+
+    const authorizedRuleTypes = await this.authorization.getAllAuthorizedRuleTypesFindOperation({
+      authorizationEntity: AlertingAuthorizationEntity.Alert,
+      ruleTypeIds: allRuleTypesIds,
+    });
+    const authorizedRuleTypesIds = Array.from(authorizedRuleTypes.keys());
+
+    const [siemRuleTypeIds, otherRuleTypeIds] = partition(authorizedRuleTypesIds, (ruleTypeId) =>
+      isSiemRuleType(ruleTypeId)
+    );
+
+    const [siemIndices, otherIndices] = await Promise.all([
+      siemRuleTypeIds ? this.getAlertIndicesAlias(siemRuleTypeIds) : [],
+      otherRuleTypeIds ? this.getAlertIndicesAlias(otherRuleTypeIds) : [],
+    ]);
+
+    const indexPatternsFetcherAsInternalUser = new IndexPatternsFetcher(this.esClient);
+    const indexPatternsFetcherAsScoped = this.esClientScoped
+      ? new IndexPatternsFetcher(this.esClientScoped)
+      : undefined;
+
+    const [{ fields: specFields }, { fields: descriptorFields }] = await Promise.all([
+      // Fetch fields for SIEM indices using current user client
+      siemIndices.length && indexPatternsFetcherAsScoped
+        ? await indexPatternsFetcherAsScoped.getFieldsForWildcard({
+            pattern: siemIndices.join(','),
+            metaFields: ['_id', '_index'],
+            fieldCapsOptions: { allow_no_indices: true },
+            includeEmptyFields: false,
+            indexFilter: INDEX_FILTER,
```

Comment:
What is this about? To exclude results from indexes/rules with low activity?

I think we should document these kinds of decisions in the function description for clarity. I would expect `getAlertFields` to return all possible alert fields I have access to(maybe I am reading this wrong though 😄 ).

And what about `includeEmptyFields`? I looked at the params type for `getFieldsForWildcard` and there is no explanation for what this does.

Reply:
These are field caps api attributes and already [documented](https://www.elastic.co/guide/en/elasticsearch/reference/8.17/search-field-caps.html)

Reply:
`INDEX_FILTER` is a range of last 90 days.
```
 {
  range: {
    '@timestamp': {
      gte: 'now-90d',
    },
  },
}
```